### PR TITLE
Add configurability 2018 roadmap

### PR DESCRIPTION
**Pull request by gregestren**
Ported from Google doc https://docs.google.com/document/d/1p7pdEwqnLOC-ATl-FdyHBJdk4Ae-5GzAYDvpJeMh4tU/edit?usp=sharing

This is an addition to @laszlocsomor's roadmap seeding with Windows (https://github.com/bazelbuild/bazel-website/pull/104)

### CODE REVIEW

**Comment by gregestren** [comment]
Note this roadmap was already shared and published in a Google doc (and the ETAs were refreshed just a week(ish) ago).

So the contents and ETAs are stable. This is mostly about porting from Google docs to a proper home.

File: roadmaps/configuration.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+
+* [Goal](#goal)
+* [Roadmap](#roadmap)
+  * [Platforms](#platforms)
+  * [User-Defined Configuration](#user-defined-configuration)
+  * [Correctness and Speed](#correctness_and_speed)
+
+
+<a name="goal"></a>
```
Comment (katre) [inline roadmaps/configuration.md]: I believe this isn't needed, and anchors are created automatically from headings in our Markdown processor.

File: roadmaps/configuration.md [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+Configurability's goal is to make Bazel a graceful multiplatform build
+tool. It also focuses on how users decide how their projects are built.
+
+
+This translates into the following high-level goals:
+
+1. **`$ bazel {build|test} :all` just works**
+    1. All targets "know" how to build for the right platforms with the right
+       toolchains and desired options
+    1. <div class="padbottom">Users only have to fiddle with options they care
```
Comment (katre) [inline roadmaps/configuration.md]: What is padbottom class doing here?

**Comment by katre** [comment]
Added @spomorski to also review.

File: roadmaps/configuration.md (hunk 1 above)
Comment (gregestren) [reply roadmaps/configuration.md]: Ah, it turns out auto-anchoring was set for [bazel](https://github.com/bazelbuild/bazel/blob/708b957f7c37fd83410588d8ef3792135abd4d8d/site/_includes/footer.html#L10-L15) but not [bazel-website](https://github.com/bazelbuild/bazel-website/blob/f8b9662c68eb022bb59c43e8c9c60b5a1683827b/_includes/footer.html).

Added it there too.

File: roadmaps/configuration.md (hunk 2 above)
Comment (gregestren) [reply roadmaps/configuration.md]: I did it so the final sublist entries aren't closer to the next entry than their own, which I thought was reader-unfriendly:

Without padding:
screen shot 2018-06-07 at 6 17 14 am

With padding:
screen shot 2018-06-07 at 6 18 58 am

I guess it's not a huge thing, so I'm happy to go with whatever the review crowd prefers.

Comment (katre) [reply roadmaps/configuration.md]: This is fine with me. Thanks for the comparative screenshots.

File: roadmaps/configuration.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+1. **"Platforms" is a first-class concept**
+    1. "Platforms" and "toolchains" are well-defined, map well to reality, and are
+        easy to create  
+    1. <div class="padbottom">Builds, build rules, remote executors, etc.
+       naturally integrate platforms</div>
+
+1. **Users decide what to configure**
+
+    `$ bazel build //myapp:fancy_edition # automatically builds my app with "fancy" features`
```
Comment (laszlocsomor) [inline roadmaps/configuration.md]: Remove the Bash comment from the command to keep the command clean. Write the comment as text above or below the command.

**Pull request merged**